tests/unit_tests/rng/include/rng_test_common.hpp

```diff
@@ -73,6 +73,24 @@ static inline bool check_equal_vector(std::vector<Fp, AllocType>& r1,
     return good;
 }
 
+template <typename Fp, typename AllocType>
+static inline bool leapfrog_check(std::vector<Fp, AllocType>& r1, std::vector<Fp, AllocType>& r2,
+                                  int n_portion, int n_engines) {
+    bool good = true;
+    int j = 0;
+    for (int i = 0; i < n_engines; i++) {
+        for (int k = 0; k < n_portion / 2; k++) {
+            for (int p = 0; p < 2; p++) {
+                if (!check_equal(r2[j++], r1[k * n_engines + i * 2 + p])) {
+                    good = false;
+                    break;
```

Comment:
I think we can replace these 2 lines with `return false` and have `return true` at the end of the function

Reply:
done